docs/includes/fundamentals/as-avs/AtlasSearchTest.php

```diff
@@ -0,0 +1,159 @@
+        ]);
+
+        Movie::insert($this->addVector([
+            ['title' => 'A', 'plot' => 'A shy teenager discovers confidence and new friendships during a transformative summer camp experience.'],
+            ['title' => 'B', 'plot' => 'A detective teams up with a hacker to unravel a global conspiracy threatening personal freedoms.'],
+            ['title' => 'C', 'plot' => 'High school friends navigate love, identity, and unexpected challenges before graduating together.'],
+            ['title' => 'D', 'plot' => 'Stranded on a distant planet, astronauts must repair their ship before supplies run out.'],
+        ]));
+
+        $moviesCollection = DB::connection('mongodb')->getCollection('movies');
+
+        try {
+            $moviesCollection->createSearchIndex([
+                'mappings' => [
+                    'fields' => [
+                        'title' => [
+                            ['type' => 'string', 'analyzer' => 'lucene.english'],
+                            ['type' => 'autocomplete', 'analyzer' => 'lucene.english'],
+                            ['type' => 'token'],
+                        ],
+                    ],
+                ],
+            ]);
+
+            $moviesCollection->createSearchIndex([
+                'mappings' => ['dynamic' => true],
+            ], ['name' => 'dynamic_search']);
+
+            $moviesCollection->createSearchIndex([
+                'fields' => [
+                    ['type' => 'vector', 'numDimensions' => 4, 'path' => 'vector4', 'similarity' => 'cosine'],
+                    ['type' => 'filter', 'path' => 'title'],
+                ],
+            ], ['name' => 'vector', 'type' => 'vectorSearch']);
+        } catch (ServerException $e) {
+            if (Builder::isAtlasSearchNotSupportedException($e)) {
+                self::markTestSkipped('Atlas Search not supported. ' . $e->getMessage());
+            }
+
+            throw $e;
+        }
+
+        // Waits for the index to be ready
+        do {
+            $ready = true;
+            usleep(10_000);
+            foreach ($collection->listSearchIndexes() as $index) {
+                if ($index['status'] !== 'READY') {
+                    $ready = false;
+                }
+            }
+        } while (! $ready);
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testSimpleSearch(): void
+    {
+        // start-search-query
+        $movies = Movie::search(
+            sort: ['title' => 1],
+            operator: Search::text('title', 'dream'),
+        )->get();
+        // end-search-query
+
+        $this->assertNotNull($movies);
+        $this->assertCount(2, $movies);
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function autocompleteSearchTest(): void
+    {
+        // start-auto-query
+        $movies = Movie::autocomplete('title', 'jak')
+            ->get();
```

Comment:
I think we can keep in a single line, it's short.
```suggestion
        $movies = Movie::autocomplete('title', 'jak')->get();
```